### PR TITLE
CloudAEye Integration with Datadog (ECOINT-19)

### PR DESCRIPTION
**Pull request by ankit-cloudaeye**
### What does this PR do?

This PR is to add CloudAEye Integration plugin in Datadog. This is for the tile-only-listing API-based integration.

### Motivation

We want to integrate our services with datadog.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [ ] If this PR includes a log pipeline, please add a description describing the remappers and processors.

### CODE REVIEW

File: cloudaeye/README.md
Comment (emarsha94) [inline cloudaeye/README.md:11]: OAuth needs to be intiated within Datadog. Please update the steps users will take to set up the integration

File: cloudaeye/manifest.json
Comment (emarsha94) [inline cloudaeye/manifest.json:42]: Add relevant tags from this list: https://docs.datadoghq.com/developers/integrations/check_references/#classifier-tags ex:
      "Queried Data Type::<FILL IN>", If pulling data from Datadog
      "Submitted Data Type::<FILL IN>" If sending in data to Datadog

File: cloudaeye/README.md
Comment (emarsha94) [reply cloudaeye/README.md:5]: 3a. what data are you sending in

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Please include an out-of-the-box dashboard with your integration, and images of the dashboard

File: cloudaeye/README.md
Comment (emarsha94) [reply cloudaeye/README.md]: This should be very clear and clear enough for a new customer understands what you product does, wht data is sent to datadog, and what you are pulling out (if applicable)

**Review by emarsha94 — CHANGES_REQUESTED** [review]

File: cloudaeye/README.md
Comment (emarsha94) [reply cloudaeye/README.md:5]: Are the events a summary of the root cause or of the incident?

**Comment by emarsha94** [comment]
👋 Quick heads up that I will be out of office next week so there may be a delay in the review from my end until July 8th. With that said, other teams will continue to review the PR. Thanks in advance for your patience!

**Review by emarsha94 — CHANGES_REQUESTED** [review]

File: cloudaeye/README.md
Comment (emarsha94) [reply cloudaeye/README.md]: This will need to be updated with the steps the user needs to take to set up the integration. You will need to remove the steps for OAuth since your integration does not use OAuth. For example, go to the CloudAEye integration in Datadog, click install, etc. This should be the exact flow the user will take to install the integration.

**Comment by emarsha94** [comment]
Hi @bhavyalatha26 - Thank you for making the changes! Your PR is missing the codeowners file. Could you please add the codeowners file to the PR? To edit via web ui: 

1. Go to your forked repository > .github > CODEOWNERS file
2. Edit the file by clicking the pencil icon in the top right 
3. Add your integration name and github handle in alphabetical order
4. Click Commit changes directly to the master branch

File: cloudaeye/README.md
Comment (emarsha94) [inline cloudaeye/README.md:5]:
```suggestion
Kosal from CloudAEye acts as your root cause co-pilot. This integration enables Datadog users to import observability data (logs, metrics, traces) and view automated root cause analyses in CloudAEye. Additionally, it identifies similar issues within Jira and related code changes that may be contributing to the incident. To expedite incident identification, Kosal sends events with a summary and root cause to Datadog.
```

File: cloudaeye/README.md
Comment (emarsha94) [inline cloudaeye/README.md:30]:
```suggestion
```

File: cloudaeye/assets/oauth_clients.json
Comment (emarsha94) [inline cloudaeye/assets/oauth_clients.json:1]: this can be deleted since you are not using oauth

File: cloudaeye/CHANGELOG.md
Comment (emarsha94) [inline cloudaeye/CHANGELOG.md:3]:
```suggestion
## 1.0.0 / 2024-07-29
```

**Review by emarsha94 — COMMENTED** [review]
Final changes then we can approve! Thank you for all of your hard work

**Review by iliakur — APPROVED** [review]

File: cloudaeye/assets/oauth_clients.json
Comment (bhavyalatha26) [reply cloudaeye/assets/oauth_clients.json:1]: done

File: cloudaeye/CHANGELOG.md
Comment (emarsha94) [reply cloudaeye/CHANGELOG.md:3]: @ankit-cloudaeye bumpoing

File: cloudaeye/README.md
Comment (emarsha94) [reply cloudaeye/README.md:30]: @ankit-cloudaeye bumpoing

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Bumping the requested changes from before then good to approve

**Comment by bhavyalatha26** [comment]
> Bumping the requested changes from before then good to approve

approved the changes. Thanks !

**Review by emarsha94 — APPROVED** [review]

File: cloudaeye/README.md
Comment (jhgilbert) [inline cloudaeye/README.md:41]:
```suggestion
3. Ensure that all API keys associated with this integration have been disabled by searching for the integration name (CloudAEye) on the [API keys management page][3].
```

**Review by jhgilbert — COMMENTED** [review]
Hi, docs team reviewer here -- I left some initial comments, but let me know when it's ready for another pass. Thank you!

File: cloudaeye/assets/dashboards/CloudAEye_overview.json
Comment (bhavyalatha26) [reply cloudaeye/assets/dashboards/CloudAEye_overview.json:1]: done

File: cloudaeye/README.md
Comment (bhavyalatha26) [reply cloudaeye/README.md:28]: done

File: cloudaeye/README.md
Comment (bhavyalatha26) [reply cloudaeye/README.md:32]: done

File: cloudaeye/README.md
Comment (bhavyalatha26) [reply cloudaeye/README.md:37]: done

File: cloudaeye/README.md
Comment (bhavyalatha26) [reply cloudaeye/README.md:19]: done

**Comment by bhavyalatha26** [comment]
Hi team , 
we have committed all your suggestion and also updated the README and cloudaeye/assets/dashboards/CloudAEye_overview.json as requested

File: cloudaeye/README.md
Comment (jhgilbert) [inline cloudaeye/README.md:9]:
```suggestion
### Pre-installation steps
```

File: cloudaeye/README.md
Comment (jhgilbert) [inline cloudaeye/README.md:28]:
```suggestion
### Notification setup
```

File: cloudaeye/README.md
Comment (jhgilbert) [inline cloudaeye/README.md:42]:
```suggestion
4. Ensure that all API keys associated with this integration have been disabled by searching for the integration name (CloudAEye) on the [API keys management page][3].
```

**Review by jhgilbert — APPROVED** [review]
Approved with a few minor suggestions, thank you!

File: cloudaeye/manifest.json
Comment (emarsha94) [inline cloudaeye/manifest.json:67]:
```suggestion
  "assets": {
    "integration": {
      "auto_install": true,
      "source_type_id":"",
      "source_type_name": "CloudAEye",
            },
      "events": {
        "creates_events": false
      },
        "dashboards": {
      "CloudAEye Kosal Dashboard": "assets/dashboards/CloudAEye_overview.json"
    }
  }
}
```

Comment (emarsha94) [reply cloudaeye/manifest.json:67]: the source type id will need to be generated for the integration. To do so, please run: 

`ddev create -t check test`

Then copy the source type id from the newly generated manifest file and paste it in this pr.

Comment (emarsha94) [reply cloudaeye/manifest.json:67]: This is so events coming in from the cloudaeye source you make in your api call to datadog will have the logo show

**Review by emarsha94 — CHANGES_REQUESTED** [review]

File: cloudaeye/manifest.json
Comment (bhavyalatha26) [reply cloudaeye/manifest.json]: Hi @emarsha94 

I get the below error when try to run the above given command:

```
Error: unknown command "create" for "ddev"
Run 'ddev --help' for usage
```

Comment (emarsha94) [reply cloudaeye/manifest.json]: please see above

File: cloudaeye/manifest.json
Comment (emarsha94) [inline cloudaeye/manifest.json:65]: Please generate the source type id based on our instructions. Thanks!

File: cloudaeye/manifest.json
Comment (emarsha94) [inline cloudaeye/manifest.json:66]:
```suggestion
      "source_type_name": "CloudAEye"
```

Comment (emarsha94) [reply cloudaeye/manifest.json:66]: @ankit-cloudaeye can you please commit this suggestion? Thanks!

File: cloudaeye/manifest.json
Comment (emarsha94) [inline cloudaeye/manifest.json:65]:
```suggestion
      "source_type_id":"26690330",
```

**Review by emarsha94 — APPROVED** [review]

File: cloudaeye/CHANGELOG.md
Comment (emarsha94) [inline cloudaeye/CHANGELOG.md:3]:
```suggestion
## 1.0.0 / 2024-10-08
```

**Comment by JoshPatel13** [comment]
Hi, could you click update branch on your PR to see if that resolves the validation issues?

**Comment by JoshPatel13** [comment]
Also, there needs to be a field called `configuration` here:

```
  "assets": {
    "integration": {
      "source_type_name": "source_type_name",
      "configuration": {},
```

**Comment by bhavyalatha26** [comment]
> Hi, could you click update branch on your PR to see if that resolves the validation issues?

Done

**Comment by bhavyalatha26** [comment]
> Also, there needs to be a field added called `configuration` here :) :
> 
> ```
>   "assets": {
>     "integration": {
>       "source_type_name": "source_type_name",
>       "configuration": {},
> ```

Ok, can you please create a commit suggestion for the same, so that we can approve directly.
TIA

File: cloudaeye/manifest.json
Comment (emarsha94) [inline cloudaeye/manifest.json:66]:
```suggestion
  "assets": {
    "integration": {
      "auto_install": true,
      "source_type_id":"26690330",
      "source_type_name": "CloudAEye"
      "configuration": {}
```

Comment (emarsha94) [reply cloudaeye/manifest.json]: This is a requirement - please commit the above.

File: cloudaeye/manifest.json
Comment (emarsha94) [inline cloudaeye/manifest.json:70]:
```suggestion
      "configuration": {},
      "events": {
        "creates_events": true
```

Comment (emarsha94) [reply cloudaeye/manifest.json:70]: @bhavyalatha26 Please see above

**Review by bhavyalatha26 — APPROVED** [review]

File: cloudaeye/manifest.json
Comment (emarsha94) [inline cloudaeye/manifest.json:75]:
```suggestion
}
}
```

Comment (emarsha94) [reply cloudaeye/manifest.json:75]: this should fix the yaml file

File: cloudaeye/manifest.json
Comment (emarsha94) [inline cloudaeye/manifest.json:72]:
```suggestion
    "dashboards": {
        "CloudAEye Kosal Dashboard": "assets/dashboards/CloudAEye_overview.json"
```

File: cloudaeye/manifest.json
Comment (JoshPatel13) [inline cloudaeye/manifest.json:73]: Dashboard needs to be moved outside the `integration` but stay inside `assets`. Thanks

**Comment by bhavyalatha26** [comment]
@JoshPatel13 I made the recommended changes, but 1 test is still failing due to some ddev issue. Can you please check and let us know the fix

**Comment by JoshPatel13** [comment]
Hi, to fix the service_checks issue, you need a `service_checks.json` in the cloudaeye directory.

**Comment by bhavyalatha26** [comment]
> Hi, to fix the service_checks issue, you need a `service_checks.json` in the cloudaeye directory.

Hi @JoshPatel13 what should be the content of that file ?

**Comment by JoshPatel13** [comment]
It can just have `[]`

**Comment by bhavyalatha26** [comment]
> It can just have `[]`

Made the changes suggested, still two more tests are failing.

Regards,
Bhavya

**Comment by JoshPatel13** [comment]
Can you try adding the file to the `assets` folder instead of it being outside of it. Also in the `manifest`, can you add 

```
"service_checks": {
        "metadata_path": "assets/service_checks.json"
      },
```
inside the `integration` object

Thanks :)

**Review by JoshPatel13 — CHANGES_REQUESTED** [review]
Requesting changes

**Comment by bhavyalatha26** [comment]
> Requesting changes

Done

File: cloudaeye/CHANGELOG.md
Comment (emarsha94) [inline cloudaeye/CHANGELOG.md:3]:
```suggestion
## 1.0.0 / 2024-11-12
```

**Review by JoshPatel13 — APPROVED** [review]
LGTM

**Pull request merged**
